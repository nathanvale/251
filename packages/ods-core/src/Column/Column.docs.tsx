/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { widthValueMap } from "./Column";
import { Columns, Column, ColumnProps, Placeholder, Stack } from "..";

export const docs: ComponentDocs<ColumnProps> = {
  category: "Layout",
  componentName: "Column",
  description:
    "Column is a low-level atomic layout component. Its is used with its parent Columns to provide 12 grid layout as well as flex Placeholder style layout.",
  propDescriptions: {
    width: "Width is used here",
  },
  migrationGuide: false,
  examples: {
    default: {
      Code: () => (
        <Columns space="small">
          <Column>
            <Placeholder height={60} />
          </Column>
          <Column>
            <Placeholder height={60} />
          </Column>
          <Column>
            <Placeholder height={60} />
          </Column>
        </Columns>
      ),
    },
    additional: [
      {
        label: "Defining Column Widths",
        description: `Bootstrap style 12 column layout can be achieved by passing in column widths.`,
        Code: () => (
          <Stack space="medium">
            {Object.keys(widthValueMap)
              .filter((key) => key !== "flex")
              .map((width) => (
                <Columns space="small" key={width}>
                  <Column width={width as any}>
                    <Placeholder height={60} label={width} />
                  </Column>
                  <Column>
                    <Placeholder height={60} />
                  </Column>
                </Columns>
              ))}
          </Stack>
        ),
      },
      {
        label: "Mixing Column Widths With Content Width and Fluid Width",
        Code: () => (
          <Columns space="small">
            <Column>
              <Placeholder />
            </Column>
            <Column width="1/3">
              <Placeholder />
            </Column>
            <Column width="content">
              <Placeholder label="content" />
            </Column>
          </Columns>
        ),
      },
      {
        label: "Stretching Space Between Columns",
        Code: () => (
          <Columns space="small">
            <Column width="content">
              <Placeholder label="content" />
            </Column>
            <Column>
              <Placeholder />
            </Column>
            <Column width="content">
              <Placeholder label="content" />
            </Column>
          </Columns>
        ),
      },
    ],
  },
  snippets: [
    {
      label: "2 Columns",
      Code: () => (
        <Columns space="small">
          <Column>
            <Stack space="small">
              <Placeholder height={60} label="Column" />
            </Stack>
          </Column>
          <Column>
            <Stack space="small">
              <Placeholder height={60} label="Column" />
            </Stack>
          </Column>
        </Columns>
      ),
    },
    {
      label: "2 Columns (Collapse Below Tablet)",
      Code: () => (
        <Columns space="small" collapseBelow="md">
          <Column>
            <Stack space="small">
              <Placeholder height={60} label="Column" />
            </Stack>
          </Column>
          <Column>
            <Stack space="small">
              <Placeholder height={60} label="Column" />
            </Stack>
          </Column>
        </Columns>
      ),
    },
    {
      label: "3 Columns",
      Code: () => (
        <Columns space="small">
          <Column>
            <Stack space="small">
              <Placeholder height={60} label="Column" />
            </Stack>
          </Column>
          <Column>
            <Stack space="small">
              <Placeholder height={60} label="Column" />
            </Stack>
          </Column>
          <Column>
            <Stack space="small">
              <Placeholder height={60} label="Column" />
            </Stack>
          </Column>
        </Columns>
      ),
    },
    {
      label: "3 Columns (Collapse Below Tablet)",
      Code: () => (
        <Columns space="small" collapseBelow="md">
          <Column>
            <Stack space="small">
              <Placeholder height={60} label="Column" />
            </Stack>
          </Column>
          <Column>
            <Stack space="small">
              <Placeholder height={60} label="Column" />
            </Stack>
          </Column>
          <Column>
            <Stack space="small">
              <Placeholder height={60} label="Column" />
            </Stack>
          </Column>
        </Columns>
      ),
    },
    {
      label: "Main Content With Sidebar",
      Code: () => (
        <Columns space="small" collapseBelow="md">
          <Column width="2/3">
            <Stack space="small">
              <Placeholder height={400} label="Main" />
            </Stack>
          </Column>
          <Column>
            <Stack space="small">
              <Placeholder height={100} label="Sidebar" />
            </Stack>
          </Column>
        </Columns>
      ),
    },
  ],
};
