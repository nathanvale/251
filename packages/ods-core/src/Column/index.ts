export { Column } from "./Column";
export type ColumnProps = import("./Column").ColumnProps;
