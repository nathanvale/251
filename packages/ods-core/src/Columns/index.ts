export { Columns, ColumnsContext } from "./Columns";
export type ColumnsProps = import("./Columns").ColumnsProps;
export type CollapsibleAlignmentChildProps = import("./Columns").CollapsibleAlignmentChildProps;
