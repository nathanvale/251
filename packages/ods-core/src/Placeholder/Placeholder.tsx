import React from "react";
import styled, { css } from "styled-components";
import { width, height, WidthProps, HeightProps } from "styled-system";
import { ResponsiveProp } from "@origin-digital/ods-types";
import { normaliseResponsiveProp } from "@origin-digital/ods-helpers";
import { Box, BoxProps } from "../Box/Box";

type PlaceholderShapeVariant = "rectangle" | "round";
export interface PlaceholderProps {
  "data-id"?: string;
  height?: ResponsiveProp<string | number>;
  label?: string;
  shape?: PlaceholderShapeVariant;
  width?: ResponsiveProp<string | number>;
}

interface ContainerProps extends WidthProps, HeightProps {
  shape: PlaceholderProps["shape"];
}

const Container = styled<Omit<BoxProps, "width" | "height">>(Box)<
  ContainerProps
>`
  ${width}
  ${height}
  border: 2px solid rgba(51, 51, 51, 0.3);
  ${p =>
    p.shape === "round"
      ? css`
          border-radius: 50%;
        `
      : undefined}}
`;

const defaultHeight = 120;
const defaultWidth = "auto";
const defaultShape = "rectangle";

export const Placeholder = ({
  "data-id": dataId,
  label,
  width = defaultWidth,
  height = defaultHeight,
  shape = defaultShape,
}: PlaceholderProps) => {
  return (
    <Container
      data-id={dataId}
      position="relative"
      overflow="hidden"
      display="flex"
      alignItems="center"
      justifyContent="center"
      width={normaliseResponsiveProp<string | number>(width)}
      height={normaliseResponsiveProp<string | number>(height)}
      shape={shape}
      backgroundColor="grey16"
    >
      {label ? (
        <Box
          paddingX="xsmall"
          paddingY="xxsmall"
          textAlign="center"
          style={{ marginTop: "2px" }}
        >
          {label}
        </Box>
      ) : (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          style={{ width: "100%", height: "100%", position: "absolute" }}
        >
          <line
            style={{ strokeWidth: "2px", stroke: "rgba(51,51,51,.1)" }}
            x1={0}
            y1={0}
            x2="100%"
            y2="100%"
          />
          <line
            style={{ strokeWidth: "2px", stroke: "rgba(51,51,51,.1)" }}
            x1="100%"
            y1={0}
            x2={0}
            y2="100%"
          />
        </svg>
      )}
    </Container>
  );
};

Placeholder.defaultProps = {
  "data-id": "placeholder",
  height: defaultHeight,
  inline: false,
  shape: defaultShape,
  width: defaultWidth,
};

Placeholder.displayName = "Placeholder";
