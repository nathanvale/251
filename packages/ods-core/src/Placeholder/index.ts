export { Placeholder } from "./Placeholder";
export type PlaceholderProps = import("./Placeholder").PlaceholderProps;
