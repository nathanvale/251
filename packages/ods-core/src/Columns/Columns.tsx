import React, { ReactElement, createContext } from "react";
import {
  ResponsiveSpace,
  CollapsibleAlignmentProps,
  OptionalTrackableProps,
  SpaceVariants,
} from "@origin-digital/ods-types";
import {
  normaliseResponsiveProp,
  mapToStyledSystem,
} from "@origin-digital/ods-helpers";
import { ColumnProps } from "../Column/Column";

import { resolveCollapsibleAlignmentProps } from "../_private/resolveCollapsibleAlignmentProps";
import { InnerContainer } from "../Inline/Inline";

export type CollapsibleAlignmentChildProps = ReturnType<
  typeof resolveCollapsibleAlignmentProps
>["collapsibleAlignmentChildProps"];

interface ColumnsContextValue {
  collapseXs: boolean;
  collapseSm: boolean;
  collapseMd: boolean;
  collapseLg: boolean;
  collapseXl: boolean;
  xsSpace?: SpaceVariants;
  smSpace?: SpaceVariants;
  mdSpace?: SpaceVariants;
  lgSpace?: SpaceVariants;
  xlSpace?: SpaceVariants;
  /**
   * The display, flexDirection and justifyContent css props per xs,sm,md,lg or
   * xl breakpoint depending on if the Columns are collapsed or aligned or not.
   * This logic is shared with the Inline component.
   */
  collapsibleAlignmentChildProps: CollapsibleAlignmentChildProps | {};
}

export const ColumnsContext = createContext<ColumnsContextValue>({
  collapseXs: false,
  collapseSm: false,
  collapseMd: false,
  collapseLg: false,
  collapseXl: false,
  xsSpace: "none",
  smSpace: "none",
  mdSpace: "none",
  lgSpace: "none",
  xlSpace: "none",
  collapsibleAlignmentChildProps: {},
});

export interface ColumnsProps
  extends CollapsibleAlignmentProps,
    OptionalTrackableProps {
  space: ResponsiveSpace;
  children:
    | Array<ReactElement<ColumnProps> | null>
    | ReactElement<ColumnProps>
    | null;
}

const defaultSpace = "none";
const defaultAlignX = "left";
const defaultAlignY = "top";

export const Columns = ({
  children,
  collapseBelow,
  "data-id": dataId,
  space = defaultSpace,
  alignX = defaultAlignX,
  alignY = defaultAlignY,
}: ColumnsProps) => {
  const [xsSpace, smSpace, mdSpace, lgSpace, xlSpace] = normaliseResponsiveProp(
    space
  );

  const {
    collapsibleAlignmentProps,
    collapsibleAlignmentChildProps,
    collapseXs,
    collapseSm,
    collapseMd,
    collapseLg,
    collapseXl,
  } = resolveCollapsibleAlignmentProps({
    collapseBelow,
    alignX,
    alignY,
  });

  return (
    <InnerContainer
      data-id={dataId}
      space={
        mapToStyledSystem({
          xs: collapseXs ? "none" : xsSpace,
          sm: collapseSm ? "none" : smSpace,
          md: collapseMd ? "none" : mdSpace,
          lg: collapseLg ? "none" : lgSpace,
          xl: collapseXl ? "none" : xlSpace,
        }) || "none"
      }
      {...collapsibleAlignmentProps}
    >
      <ColumnsContext.Provider
        value={{
          collapseXs,
          collapseSm,
          collapseMd,
          collapseLg,
          collapseXl,
          xsSpace,
          smSpace,
          mdSpace,
          lgSpace,
          xlSpace,
          collapsibleAlignmentChildProps,
        }}
      >
        {children}
      </ColumnsContext.Provider>
    </InnerContainer>
  );
};

Columns.defaultProps = {
  "data-id": "columns",
  space: "large",
  alignY: defaultAlignY,
  alignX: defaultAlignX,
};

Columns.displayName = "Columns";
