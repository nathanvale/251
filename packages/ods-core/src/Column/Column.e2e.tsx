import * as React from "react";
import { E2ETests } from "@origin-digital/ods-types";
import { Placeholder } from "../Placeholder";
import { Stack } from "../Stack";
import { Columns } from "../Columns";
import { Heading } from "../Heading";
import { Column, widthValueMap } from "./Column";

export const tests: E2ETests = [
  {
    label: "AlignY Bottom",
    Code: () => (
      <Stack space="medium">
        {Object.keys(widthValueMap)
          .filter((key) => key !== "flex")
          .map((width) => (
            <Columns space="small" key={width}>
              <Column width={width as any}>
                <Placeholder height={60} label={width} />
              </Column>
              <Column>
                <Placeholder height={60} />
              </Column>
            </Columns>
          ))}
      </Stack>
    ),
  },
  {
    label: "Mixing Column Widths With Content Width and Fluid Width",
    Code: () => (
      <Columns space="small">
        <Column>
          <Placeholder />
        </Column>
        <Column width="1/3">
          <Placeholder />
        </Column>
        <Column width="content">
          <Placeholder label="content" />
        </Column>
      </Columns>
    ),
  },
  {
    label: "Stretching Space Between Columns",
    Code: () => (
      <Columns space="small">
        <Column width="content">
          <Placeholder label="content" />
        </Column>
        <Column>
          <Placeholder />
        </Column>
        <Column width="content">
          <Placeholder label="content" />
        </Column>
      </Columns>
    ),
  },
  {
    label: "Horizontal alignment",
    responsive: true,
    Code: () => (
      <Columns>
        <Column>
          <Placeholder />
        </Column>
        <Column alignX={["center", "right"]}>
          <Placeholder width="50px" height="50px" />
        </Column>
      </Columns>
    ),
  },
  {
    label: "Column display block when alignX = left",
    Code: () => (
      <Columns>
        <Column>
          <Heading variant="h4">Amount paid</Heading>
        </Column>
        <Column>
          <Heading data-id="amount" variant="h2" align="right">
            $25.00
          </Heading>
        </Column>
      </Columns>
    ),
  },
];
