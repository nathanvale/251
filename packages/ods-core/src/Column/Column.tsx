/* eslint-disable no-eval */
import React, { ReactNode, useContext } from "react";
import styled, { css } from "styled-components";
import {
  mapSpaceAliasToIndex,
  setBreakpoint,
} from "@origin-digital/ods-helpers";
import { ColumnsContext } from "../Columns/Columns";
import { Box } from "../Box/Box";
import { BoxDebug } from "../_private/components/BoxDebug/BoxDebug";

export interface ColumnProps {
  children: ReactNode;
  width?:
    | "content"
    | "1/2"
    | "1/3"
    | "2/3"
    | "1/4"
    | "3/4"
    | "1/6"
    | "5/6"
    | "1/12"
    | "5/12"
    | "7/12"
    | "11/12"
    | "flex";
  "data-id"?: string;
}

const OuterStyledBox = styled(Box)<{ columnWidth: ColumnProps["width"] }>`
  min-width: 0;
  ${p => (p.columnWidth === "content" ? "flex-shrink: 0;" : undefined)}
  ${p =>
    p.columnWidth && p.columnWidth !== "content" && p.columnWidth !== "flex"
      ? css`
          flex: 0 0 ${eval(p.columnWidth) * 100}%;
        `
      : undefined}
`;

const InnerStyledBox = styled(Box)`
  div:first-child > & {
    padding-top: 0;
  }
`;

export const Column = ({
  children,
  width = "flex",
  "data-id": dataId,
}: ColumnProps) => {
  const { collapseBelow, space } = useContext(ColumnsContext);
  const spaceIndex = space && mapSpaceAliasToIndex({ space });
  return (
    <OuterStyledBox
      data-id={dataId}
      columnWidth={width}
      width={width !== "content" ? "full" : undefined}
    >
      <InnerStyledBox
        height="full"
        paddingLeft={
          (collapseBelow
            ? setBreakpoint({
                breakpoint: collapseBelow,
                value1: "none",
                value2: spaceIndex,
              })
            : spaceIndex) as TS_FIXME
        }
        paddingTop={
          (collapseBelow
            ? setBreakpoint({
                breakpoint: collapseBelow,
                value1: spaceIndex,
                value2: "none",
              })
            : "none") as TS_FIXME
        }
      >
        <BoxDebug>{children}</BoxDebug>
      </InnerStyledBox>
    </OuterStyledBox>
  );
};

Column.defaultProps = {
  "data-id": "column",
  width: "flex",
};

Column.displayName = "Column";
