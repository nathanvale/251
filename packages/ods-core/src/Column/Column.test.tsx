import React from "react";
import {render} from "@origin-digital/ods-testing-library";
import {Columns} from "../Columns/Columns";
import {Placeholder} from "../Placeholder/Placeholder";
import {Column} from "./Column";

test("it sets width to flex by default", () => {
  const {container} = render(
    <Columns space="small">
      <Column>
        <Placeholder />
      </Column>
      <Column>
        <Placeholder />
      </Column>
      <Column>
        <Placeholder />
      </Column>
    </Columns>,
  );
  expect(container.firstChild).toMatchSnapshot();
});

test('it correctly allocates required width for "content"', () => {
  const {container} = render(
    <Columns space="small">
      <Column width="content">
        <Placeholder>This is content width</Placeholder>
      </Column>
      <Column>
        <Placeholder />
      </Column>
      <Column>
        <Placeholder />
      </Column>
    </Columns>,
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("it correctly sets width for proportions 1/3 and 2/3", () => {
  const {container} = render(
    <Columns space="small">
      <Column width="1/3">
        <Placeholder>This is 1/3</Placeholder>
      </Column>
      <Column width="2/3">
        <Placeholder>This is 2/3</Placeholder>
      </Column>
    </Columns>,
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("it correctly sets width for proportions 1/12, 7/12, 1/3", () => {
  const {container} = render(
    <Columns space="small">
      <Column width="1/12">
        <Placeholder>This is 1/12</Placeholder>
      </Column>
      <Column width="5/12">
        <Placeholder>This is 5/12</Placeholder>
      </Column>
      <Column width="1/3">
        <Placeholder>This is 1/3</Placeholder>
      </Column>
    </Columns>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
