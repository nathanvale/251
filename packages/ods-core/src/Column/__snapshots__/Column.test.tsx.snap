// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it correctly allocates required width for "content" 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: -12px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c9 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  width: 100%;
}

.c4 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  padding-top: 0;
  padding-left: 12px;
  height: 100%;
}

.c2 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.c5 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 24px;
  padding-right: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
}

.c8 {
  min-width: 0;
}

.c1 {
  min-width: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

div:first-child > .c3 {
  padding-top: 0;
}

.c6 {
  font-size: 16px;
  line-height: 24px;
  -webkit-transform: translateY(0.36em);
  -ms-transform: translateY(0.36em);
  transform: translateY(0.36em);
  padding-top: 1px;
  font-family: gordita,sans-serif;
  font-weight: 500;
  color: #232323;
}

.c6:before {
  content: "";
  margin-top: -13px;
  display: block;
  height: 0;
}

<div
  class="c0"
  data-id="columns"
  display="flex"
>
  <div
    class="c1 c2"
    data-id="column"
  >
    <div
      class="c3 c4"
      height="full"
    >
      <div
        class="c2"
      >
        <div
          class="c5"
          data-id="placeholder"
          display="flex"
        >
          <span
            class="c6 -span c7"
            color="grey56"
            data-id="text"
            display="block"
          >
            This is content width
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c8 c9"
    data-id="column"
    width="full"
  >
    <div
      class="c3 c4"
      height="full"
    >
      <div
        class="c2"
      >
        <div
          class="c5"
          data-id="placeholder"
          display="flex"
        >
          <span
            class="c6 -span c7"
            color="grey56"
            data-id="text"
            display="block"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="c8 c9"
    data-id="column"
    width="full"
  >
    <div
      class="c3 c4"
      height="full"
    >
      <div
        class="c2"
      >
        <div
          class="c5"
          data-id="placeholder"
          display="flex"
        >
          <span
            class="c6 -span c7"
            color="grey56"
            data-id="text"
            display="block"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it correctly sets width for proportions 1/3 and 2/3 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: -12px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  width: 100%;
}

.c4 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  padding-top: 0;
  padding-left: 12px;
  height: 100%;
}

.c5 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.c6 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 24px;
  padding-right: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
}

.c1 {
  min-width: 0;
  -webkit-flex: 0 0 33.33333333333333%;
  -ms-flex: 0 0 33.33333333333333%;
  flex: 0 0 33.33333333333333%;
}

.c9 {
  min-width: 0;
  -webkit-flex: 0 0 66.66666666666666%;
  -ms-flex: 0 0 66.66666666666666%;
  flex: 0 0 66.66666666666666%;
}

div:first-child > .c3 {
  padding-top: 0;
}

.c7 {
  font-size: 16px;
  line-height: 24px;
  -webkit-transform: translateY(0.36em);
  -ms-transform: translateY(0.36em);
  transform: translateY(0.36em);
  padding-top: 1px;
  font-family: gordita,sans-serif;
  font-weight: 500;
  color: #232323;
}

.c7:before {
  content: "";
  margin-top: -13px;
  display: block;
  height: 0;
}

<div
  class="c0"
  data-id="columns"
  display="flex"
>
  <div
    class="c1 c2"
    data-id="column"
    width="full"
  >
    <div
      class="c3 c4"
      height="full"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
          data-id="placeholder"
          display="flex"
        >
          <span
            class="c7 -span c8"
            color="grey56"
            data-id="text"
            display="block"
          >
            This is 1/3
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c9 c2"
    data-id="column"
    width="full"
  >
    <div
      class="c3 c4"
      height="full"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
          data-id="placeholder"
          display="flex"
        >
          <span
            class="c7 -span c8"
            color="grey56"
            data-id="text"
            display="block"
          >
            This is 2/3
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it correctly sets width for proportions 1/12, 7/12, 1/3 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: -12px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  width: 100%;
}

.c4 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  padding-top: 0;
  padding-left: 12px;
  height: 100%;
}

.c5 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.c6 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 24px;
  padding-right: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
}

.c10 {
  min-width: 0;
  -webkit-flex: 0 0 33.33333333333333%;
  -ms-flex: 0 0 33.33333333333333%;
  flex: 0 0 33.33333333333333%;
}

.c1 {
  min-width: 0;
  -webkit-flex: 0 0 8.333333333333332%;
  -ms-flex: 0 0 8.333333333333332%;
  flex: 0 0 8.333333333333332%;
}

.c9 {
  min-width: 0;
  -webkit-flex: 0 0 41.66666666666667%;
  -ms-flex: 0 0 41.66666666666667%;
  flex: 0 0 41.66666666666667%;
}

div:first-child > .c3 {
  padding-top: 0;
}

.c7 {
  font-size: 16px;
  line-height: 24px;
  -webkit-transform: translateY(0.36em);
  -ms-transform: translateY(0.36em);
  transform: translateY(0.36em);
  padding-top: 1px;
  font-family: gordita,sans-serif;
  font-weight: 500;
  color: #232323;
}

.c7:before {
  content: "";
  margin-top: -13px;
  display: block;
  height: 0;
}

<div
  class="c0"
  data-id="columns"
  display="flex"
>
  <div
    class="c1 c2"
    data-id="column"
    width="full"
  >
    <div
      class="c3 c4"
      height="full"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
          data-id="placeholder"
          display="flex"
        >
          <span
            class="c7 -span c8"
            color="grey56"
            data-id="text"
            display="block"
          >
            This is 1/12
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c9 c2"
    data-id="column"
    width="full"
  >
    <div
      class="c3 c4"
      height="full"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
          data-id="placeholder"
          display="flex"
        >
          <span
            class="c7 -span c8"
            color="grey56"
            data-id="text"
            display="block"
          >
            This is 5/12
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c10 c2"
    data-id="column"
    width="full"
  >
    <div
      class="c3 c4"
      height="full"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
          data-id="placeholder"
          display="flex"
        >
          <span
            class="c7 -span c8"
            color="grey56"
            data-id="text"
            display="block"
          >
            This is 1/3
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it sets width to flex by default 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: -12px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  width: 100%;
}

.c4 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  padding-top: 0;
  padding-left: 12px;
  height: 100%;
}

.c5 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.c6 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 24px;
  padding-right: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
}

.c1 {
  min-width: 0;
}

div:first-child > .c3 {
  padding-top: 0;
}

.c7 {
  font-size: 16px;
  line-height: 24px;
  -webkit-transform: translateY(0.36em);
  -ms-transform: translateY(0.36em);
  transform: translateY(0.36em);
  padding-top: 1px;
  font-family: gordita,sans-serif;
  font-weight: 500;
  color: #232323;
}

.c7:before {
  content: "";
  margin-top: -13px;
  display: block;
  height: 0;
}

<div
  class="c0"
  data-id="columns"
  display="flex"
>
  <div
    class="c1 c2"
    data-id="column"
    width="full"
  >
    <div
      class="c3 c4"
      height="full"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
          data-id="placeholder"
          display="flex"
        >
          <span
            class="c7 -span c8"
            color="grey56"
            data-id="text"
            display="block"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="c1 c2"
    data-id="column"
    width="full"
  >
    <div
      class="c3 c4"
      height="full"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
          data-id="placeholder"
          display="flex"
        >
          <span
            class="c7 -span c8"
            color="grey56"
            data-id="text"
            display="block"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="c1 c2"
    data-id="column"
    width="full"
  >
    <div
      class="c3 c4"
      height="full"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
          data-id="placeholder"
          display="flex"
        >
          <span
            class="c7 -span c8"
            color="grey56"
            data-id="text"
            display="block"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
